src/components/Navbar.tsx

```diff
@@ -0,0 +1,105 @@
+import { FaUserCircle } from "react-icons/fa";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { clearAuth } from "../modules/auth/authSlice";
+import AuthLogo from "./AuthLogo";
+import ThemeToggle from "../modules/theme/ThemeToggle";
+
+const Navbar = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  // Handle logout
+  const handleLogout = () => {
+    dispatch(clearAuth());
+    localStorage.removeItem("accessToken");
+    localStorage.removeItem("refreshToken");
+    localStorage.removeItem("profile");
+    navigate("/auth");
+  };
+
+  // Show or hide the modal
+  const toggleModal = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+
+  // Toggle dropdown
+  const toggleDropdown = () => {
+    setIsDropdownOpen(!isDropdownOpen);
+  };
+
+  return (
+    <div className="navbar flex bg-gray-300 dark:bg-darkSurface text-primaryDark dark:text-lightText py-5 px-1">
+      <div className="flex-1">
+        <AuthLogo />
+      </div>
+
+      <div className="relative">
+        {/* User Avatar Button */}
+        <div className="flex items-center">
+          <button
+            onClick={toggleDropdown}
+            className="btn btn-ghost btn-circle avatar"
+          >
+            <div className="w-10 rounded-full">
+              <FaUserCircle className="text-4xl text-primaryDark dark:text-lightText" />
+            </div>
+          </button>
+          <ThemeToggle />
+        </div>
+
+        {/* Dropdown Menu (hidden by default) */}
+        {isDropdownOpen && (
+          <ul className="absolute right-0 mt-5 z-10 menu menu-sm dropdown-content bg-gray-300 dark:bg-darkSurface rounded-box w-52 p-2 shadow-lg">
+            <li className="hover:dark:bg-gray-800 hover:bg-slate-200  rounded-sm cursor-pointer p-2">
+              <a className="justify-between text-primaryDark dark:text-lightText">
+                Profile
+                <span className="badge">New</span>
+              </a>
+            </li>
+
+            <li
+              className="hover:dark:bg-gray-800 hover:bg-slate-200 rounded-sm cursor-pointer p-2"
+              onClick={toggleModal}
+            >
+              <a className="text-primaryDark dark:text-lightText">Logout</a>
```

Comment:
This should be a button and take the onClick on the list item.

Reply:
how it would behave different than the current one ?